zeus/optimizer/power_limit.py

```diff
@@ -468,3 +472,93 @@ def _save_profile(self) -> None:
                 ),
             )
         self.logger.info("JIT profiling results saved to '%s'.", str(self.profile_path))
+
+
+# Only import HuggingFace Classes when type checking, to avoid hard dependency on HuggingFace Transformers
+if TYPE_CHECKING:
+    from transformers import (
+        TrainerCallback,
+        TrainingArguments,
+        TrainerState,
+        TrainerControl,
+        PreTrainedModel,
+    )
+
+try:
+    from transformers import TrainerCallback
+
+    transformers_available = True
+except ModuleNotFoundError:
+    transformers_available = False
+    TrainerCallback = object  # Fallback base class
+
+
+class HFGlobalPowerLimitOptimizer(TrainerCallback):
+    """[Wrapped for Hugging Face Trainer Callback] Optimizer for the power limit knob.
+
+    This optimizer uses the JIT profiling log to determine the optimal power limit.
+    """
+
+    def __init__(
+        self,
+        monitor: ZeusMonitor,
+        optimum_selector: OptimumSelector | None = None,
+        wait_steps: int = 1,
+        warmup_steps: int = 10,
+        profile_steps: int = 40,
+        pl_step: int = 25,
+        profile_path: str | Path | None = None,
+    ) -> None:
+        r"""[Wrapped for Hugging Face Trainer Callback] Initialize the optimizer.
```

Comment:
I think you can have the bracket just in the class docstring. Please remove from other methods.